### PR TITLE
Improve SoList hiding

### PR DESCRIPTION
**Pull request by JingMatrix**
## Changes

1.  Export CMAKE symbols for `clangd` language server so that current project is more friendly for non-IDE users.
2. Fix linker symbol lookup on Android 15 for some devices (#63 )
3. Drop module SoInfo from SoList to avoid detection

## Why 

In Bionic linker, the SoInfo structure has a field `next`, which points to the next loaded library in a linked list consisting of all loaded libraries.
Hence, a hooked app can easily find all loaded libraries.
Previously in ReZygisk, module library items are hidden by setting the `pathname` to be empty. This was futile for two reasons:
1. The `name` field still keeps the library name
2. An item with empty name or pathname is artificial and can be used to detect Zygisk.

By contrast, dropping the item from SoList is a widely used methods in many projects, such as the well-known Riru.
You can search keyword `solist_remove_soinfo` on GitHub to find them.

## Checkmarks

- [x] The modified functions have been tested.
- [x] Used the same indentation as the rest of the project.
- [ ] Updated documentation (changelog).

## Additional information

If you have any additional information, write it here

### CODE REVIEW

**Comment by JingMatrix** [comment]
To get a clear idea, this pull-request will log new items like this:
```
[ 2024-11-29T13:47:30.494    10242: 16356: 16356 I/zygisk-core64   ] Dropped SoList entry for liblspd.so loaded at /memfd:jit-cache-zygisk (deleted)
```

**Comment by JingMatrix** [comment]
Very interestingly, we cannot drop the soinfo of `libzygisk.so` loaded from `/data/adb/rezygisk/lib64/libzygisk.so`.
Otherwise, zygisk modules will fail to work, a testing commit is https://github.com/JingMatrix/ReZygisk/commit/c5ac305354f6099cbfc3fb7ae65453d5a2782d92

**Comment by JingMatrix** [comment]
I made an app to detect ReZygisk: [JingMatrix/Demo](https://github.com/JingMatrix/Demo).
Here are the results:
1. No versions of ReZygisk ever passed the second test of `solist`;
2. Current PR will help to pass the first test of `solist`

**Comment by JingMatrix** [comment]
Current pull-request can be merged.

To bypass the test of Demo, here is idea of injecting `libzygisk.so` to be implemented later :
1. In daemon, remote call dlopen to load `libzygisk.so`
2. In `libzygisk.so`, move/copy its memory block to a new place, and return the new address back to daemon
3. In daemon, remote call dlclose to clean all traces introduced by the dynamic linker, including the one of `solist`
4. In daemon, remote call entry function using the new address returned in step 2

Basically, we now move the memory block of `libzygisk.so` instead of simply renaming it.
